### PR TITLE
Remove Solaris conditions and use PYTHONHOME only when set

### PR DESCRIPTION
**Pull request by larham**
- Remove conditionals for Solaris
- Only use PYTHONHOME when necessary

### CODE REVIEW

File: gpMgmt/bin/generate-greenplum-path.sh [hunk 1]
```diff
@@ -57,36 +57,29 @@ cat <<EOF
...
+# OSX does not need JAVA_HOME
 if [ "${PLAT}" = "Darwin" ] ; then
 cat << EOF
-DYLD_LIBRARY_PATH=\$GPHOME/lib:\$PYTHONHOME/lib:\$DYLD_LIBRARY_PATH
+DYLD_LIBRARY_PATH=${TEMP_LIB_STR}:\${DYLD_LIBRARY_PATH-}
 EOF
 fi
 
 # OSX does not have LD_LIBRARY_PATH
 if [ "${PLAT}" != "Darwin" ] ; then
```
[the remark this answers is not in the document]
Comment (pivotal-mike) [reply gpMgmt/bin/generate-greenplum-path.sh]: @larham @danielgustafsson Fun fact -MacOS also support LD_LIBRARY_PATH for dylib discovery: ([https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/UsingDynamicLibraries.html](Dev. docs ref)).  We may be at the point where needing to distinguish Mac v. Linux is no longer needed

**Comment by danielgustafsson** [comment]
> On 01 Nov 2017, at 14:38, Michael Roth <notifications@github.com> wrote:

> @larham @danielgustafsson Fun fact -MacOS also support LD_LIBRARY_PATH for dylib discovery: ([https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/UsingDynamicLibraries.html](Dev. docs ref)). We may be at the point where needing to distinguish Mac v. Linux is no longer needed

Cool, did not know that. I agree, we should investigate this and make this PR even better.

cheers ./daniel

**Comment by larham** [comment]
Thanks, all, for the feedback, which I think we've adopted 100%. Please review, and approve if you see fit.

File: gpMgmt/bin/generate-greenplum-path.sh [hunk 2]
```diff
@@ -57,37 +57,21 @@ cat <<EOF
...
-LD_LIBRARY_PATH=\$GPHOME/lib:\$PYTHONHOME/lib:/usr/sfw/lib:/usr/local/python/lib:\$LD_LIBRARY_PATH
+cat <<EOF
+PATH=${GP_BIN_PATH}:\$PATH
 EOF
-    else
-    cat <<EOF
-LD_LIBRARY_PATH=\$GPHOME/lib:\$PYTHONHOME/lib:\$LD_LIBRARY_PATH
+
+cat <<EOF
+LD_LIBRARY_PATH=${GP_LIB_PATH}:\${LD_LIBRARY_PATH-}
```
Comment (Chibin) [inline gpMgmt/bin/generate-greenplum-path.sh]: Just curious, what does the dash do in `${LD_LIBRARY_PATH-}`? Does this just means our default value is blank?

Comment (larham) [reply gpMgmt/bin/generate-greenplum-path.sh]: Right, default is null/unset, such that the resulting string is empty.

**Pull request merged**